### PR TITLE
Obtain system's boot ID and machine ID

### PR DESCRIPTION
**Pull request by ejoerns**
For now, we just collect some information to be used in follow-up features:

* machine-id -> allows to identify the machine we run on
* boot-id -> allows to identify the boot we are in

Related to #1070 and #1114

### CODE REVIEW

File: src/context.c [hunk 1]
```diff
@@ -321,6 +335,9 @@ static gboolean r_context_configure_target(GError **error)
 		context->bootslot = get_cmdline_bootname();
 	}
 
+	context->boot_id = get_boot_id();
```
Comment (jluebbe) [inline src/context.c:352]: This leaks on reconfiguration. Add `g_clear_pointer()`.

File: src/context.c [hunk 2]
```diff
@@ -27,6 +27,20 @@ static gchar *regex_match(const gchar *pattern, const gchar *string)
 	return NULL;
 }
 
+static gchar* get_machine_id(void)
+{
+	gchar *contents = NULL;
+	g_autoptr(GError) ierror = NULL;
+
+	if (!g_file_get_contents("/etc/machine-id", &contents, NULL, &ierror)) {
+		g_warning("Failed to get machine-id: %s", ierror->message);
```
Comment (jluebbe) [inline src/context.c:36]: This probably shouldn't be a `g_warning` (perhaps `g_info`), as not all targets use systemd. It also breaks the tests, as warning are fatal there.

File: src/context.c [hunk 3]
```diff
@@ -321,6 +349,11 @@ static gboolean r_context_configure_target(GError **error)
 		context->bootslot = get_cmdline_bootname();
 	}
 
+	context->boot_id = get_boot_id();
+	g_debug("Obtained system boot ID: '%s'", context->boot_id);
+	context->machine_id = get_machine_id();
```
Comment (jluebbe) [inline src/context.c:354]: This leaks as well on reconfiguration.

**Review by jluebbe — CHANGES_REQUESTED** [review]

File: src/context.c (hunk 1 above)
Comment (ejoerns) [reply src/context.c:355]: Fixed.

File: src/context.c (hunk 2 above)
Comment (ejoerns) [reply src/context.c]: Ack, fixed.

**Review by jluebbe — APPROVED** [review]

**Pull request merged**